Changelog.md

```diff
@@ -2,6 +2,7 @@
 
 Language Features:
  * Ability to select the abi coder using ``pragma abicoder v1`` and ``pragma abicoder v2``.
+ * Inline Assembly: Use ``.offset`` and ``.length`` for calldata variables of dynamic array type to access their calldata offset and length (number of elements). Both of them can also be assigned to.
```

Comment:
What is happening on assignment?

Reply:
As always, it just resets the stack slot contents - just like with `x.slot` for storage pointers.

Reply:
So the use case here would be that functions can modify to which part of the calldata something points to before passing those calldata variables further down.

Perhaps this could be useful for some proxy contracts removing some incoming wrapping?

Reply:
Another use case is to do something like
```
function sig() returns (bytes calldata s) {
assembly { s.offset := 0 s.length := 4 }
}
```